aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -52,6 +52,24 @@ static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, b
   return r;
 }
 
+Tensor autocast_to_fp16(const Tensor& self) {
+  if (self.dtype() == at::ScalarType::Float) {
```

Comment:
You could use or borrow from the [`is_eligible`](https://github.com/csarofeen/pytorch/blob/9376306a4306295a3559d30c335aba6d7a753996/aten/src/ATen/autocast_mode.h#L73) helper in autocast_mode.h.

Reply:
> To fully imitate the eager policy you could use or borrow from the [`is_eligible`](https://github.com/csarofeen/pytorch/blob/9376306a4306295a3559d30c335aba6d7a753996/aten/src/ATen/autocast_mode.h#L73) helper in autocast_mode.h.

Right. The `is_cuda` part is TBD - I'd love to understand better the rationale and how relevant it is to the JIT implementation. (since this is still work in progress I'm not too concerned about artificial policy tweaks - we can do that any time before we upstream)

Is there any other part of the logic that's missing?